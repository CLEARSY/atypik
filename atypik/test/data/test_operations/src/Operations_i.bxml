<?xml version="1.0" encoding="UTF-8" ?>
<Machine xmlns='https://www.atelierb.eu/Formats/bxml' version='1.0' name='Operations_i' type='implementation' position='true' semantic='true' b0check='false'>
<Abstraction>Operations</Abstraction>
<Initialisation>
<Skip/>
</Initialisation>
<Operations>
<Operation name='pair'>
<Attr>
<Pos l='13' c='5' s='4' endLine='17'/>
</Attr>
<Output_Parameters>
<Id value='out1' typref='1'>
<Attr>
<Pos l='13' c='5' s='4'/>
</Attr>
</Id>
<Id value='out2' typref='1'>
<Attr>
<Pos l='13' c='11' s='4'/>
</Attr>
</Id>
</Output_Parameters>
<Input_Parameters>
<Id value='in1' typref='1'>
<Attr>
<Pos l='13' c='25' s='3'/>
</Attr>
</Id>
<Id value='in2' typref='1'>
<Attr>
<Pos l='13' c='30' s='3'/>
</Attr>
</Id>
</Input_Parameters>
<Body>
<Nary_Sub op=';'>
<Assignement_Sub>
<Attr>
<Pos l='15' c='14' s='2'/>
</Attr>
<Variables>
<Id value='out1' typref='1'>
<Attr>
<Pos l='15' c='9' s='4'/>
</Attr>
</Id>
</Variables>
<Values>
<Id value='in1' typref='1'>
<Attr>
<Pos l='15' c='17' s='3'/>
</Attr>
</Id>
</Values>
</Assignement_Sub>
<Assignement_Sub>
<Attr>
<Pos l='16' c='14' s='2'/>
</Attr>
<Variables>
<Id value='out2' typref='1'>
<Attr>
<Pos l='16' c='9' s='4'/>
</Attr>
</Id>
</Variables>
<Values>
<Id value='in2' typref='1'>
<Attr>
<Pos l='16' c='17' s='3'/>
</Attr>
</Id>
</Values>
</Assignement_Sub>
</Nary_Sub>
</Body>
</Operation>
</Operations>
<TypeInfos>
<Type id='0'>
<Id value='BOOL'/>
</Type>
<Type id='1'>
<Id value='INTEGER'/>
</Type>
<Type id='2'>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Type>
<Type id='3'>
<Unary_Exp op='POW'>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Unary_Exp>
</Type>
</TypeInfos>

</Machine>