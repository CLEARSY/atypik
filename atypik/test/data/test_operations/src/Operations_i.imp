/* Operations_i
* Author: vdemeulenaere
* Creation date: 13/06/2023
*/

IMPLEMENTATION Operations_i
REFINES Operations


    
OPERATIONS 
    
    out1, out2 <-- pair(in1, in2) =
    BEGIN
        out1 := in1;
        out2 := in2    
    END
    
    
END