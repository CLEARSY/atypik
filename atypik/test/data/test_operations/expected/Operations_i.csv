Expression	Identifier	Line	Column	Type
"in1"	8	13	25	INTEGER
"out1"	6	15	9	INTEGER
"out2"	7	16	9	INTEGER
"in2"	9	16	17	INTEGER
